migrations/src/postDeploymentSetup.js

```diff
@@ -44,23 +44,28 @@ module.exports = async () => {
   // Only transfer tokens in testnets
   if (NETWORK !== 'mainnet') {
     // Add new instance of StakeManager contract & Deployer address as Minter
-    await schellingCoin.addMinter(stakeManagerAddress);
-    await schellingCoin.addMinter(signers[0].address);
 
-    if (SCHELLING_COIN_ADDRESS !== '') {
-      const { StakeManager: oldStakeManagerAddress } = await readOldDeploymentFile();
+    const initialSupply = await schellingCoin.INITIAL_SUPPLY();
+    const stakeManagerSupply = (BigNumber.from(10).pow(BigNumber.from(26))).mul(BigNumber.from(6));
 
+    if (SCHELLING_COIN_ADDRESS !== '') {
       // if previous instances of Schelling Coin is reused again and again,
       // then initial balance will get depleted, thus intial tokens minting is needed,
       // each time Schelling Coin instance is reused
-      const initialSupply = await schellingCoin.INITIAL_SUPPLY();
-      await schellingCoin.mint(signers[0].address, initialSupply);
-
-      // Remove previous instance of StakeManager contract & Deployer address from Minter
-      await schellingCoin.removeMinter(oldStakeManagerAddress);
+      await schellingCoin.addMinter(signers[0].address);
+      if (schellingCoin.balanceOf(signers[0].address) === 0) {
```

Comment:
Hey Why do you think this condn is needed
As per me
```
 await schellingCoin.mint(signers[0].address, initialSupply - schellingCoin.balanceOf(signers[0].address));
 ```
 
Is all we need, if signers[0] balance is 0, mint Initial Supply - 0 == Intial supply 
Thats expected behav

Reply:
yeah! Agreed!